src/rewrites.rs

```diff
@@ -116,42 +116,49 @@ pub fn rules_from_str(rs: Vec<&str>) -> Vec<Rewrite<Mdl, TensorAnalysis>> {
 
 /// Struct for passing results in the recursive function check_pat
 ///
-/// Similar as ValTnsr for TensorAnalysis, but with tnsr being the object 
-/// rather than pointer, to make memory working correctly with recursive 
+/// Similar as ValTnsr for TensorAnalysis, but with tnsr being the object
+/// rather than pointer, to make memory working correctly with recursive
```

Comment:
How does recursion come into play here?

Reply:
check_pat function is recursive. This struct is for the results passing in that function